game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1979,6 +1979,27 @@ private void updateStepFromEdt() {
     }
   }
 
+  private @Nullable Collection<UnitCategory> getUpdatedUnitsToPlace(
+      final PlayerId lastPlayer, final PlayerId player, final GameStep step) {
+    if (!ClientSetting.showBetaFeatures.getValueOrThrow()) {
+      return null;
+    }
+    Collection<UnitCategory> unitCategories = null;
+    // Keep track if we're past the production step for the current player.
+    // If the current player changes, reset to false.
+    if (player == null || !player.equals(lastPlayer)) {
+      postProductionStep = false;
```

Comment:
I'm a bit nervous about keeping state in `TripleAFrame`. Have you had any thoughts about extracting this logic to its own class @asvitkine ? It's a medium/long term plan to refactor TripleAFrame to not be quite the 'everything' class, it has too many things it does.